Floofbot/Modules/Administration.cs

```diff
@@ -473,7 +477,7 @@ public async Task warnlog([Summary("user")] string user)
                     if (warning.Forgiven)
                     {
                         IUser forgivenBy = resolveUser(warning.ForgivenBy.ToString());
-                        var forgivenByText = (forgivenBy != null) ? "" : $"(forgiven by {forgivenBy.Username}#{forgivenBy.Discriminator}";
+                        var forgivenByText = (forgivenBy == null) ? "" : $"(forgiven by {forgivenBy.Username}#{forgivenBy.Discriminator})";
```

Comment:
In the below functions, `forgivenBy` is always set when `Forgiven` is `true`. Is it actually possible for it to be `null`?

Reply:
If the moderator that forgave the user is no longer in the server, the function will not be able to resolve the user ID, so the function will return a null result